rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -275,6 +277,27 @@ public Repository getRepository() {
         return repository;
     }
 
+    public BlockExecutorFactory getBlockExecutorFactory() {
+        if (blockExecutorFactory == null) {
+            blockExecutorFactory = buildBlockExecutorFactory();
+        }
+
+        return blockExecutorFactory;
+    }
+
+    private BlockExecutorFactory buildBlockExecutorFactory() {
+        return new BlockExecutorFactory(
+                getRskSystemProperties(),
+                getRepository(),
+                getBlockStore(),
+                getReceiptStore(),
+                getProgramInvokeFactory(),
+                getCompositeEthereumListener(),
```

Comment:
By using this listener, the `debug_traceTransaction` will propagate events to real listeners (like `TransactionPoolImpl`). I think this could make other components fail.

Reply:
Done